### PR TITLE
Bug 1152173 - [Stingray][Home] Implement folder list

### PR DESCRIPTION
**Pull request by rexboy7**
This bug mainly do:
- Add folder list to home.
- Expands folder list after focusing a folder.
- Let `scrollable` able to specify reference element as an 'anchor point' to determine positions of its nodes.

Since edit is not completed yet, please use `defaule-cards.json` for reviewing.

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c38c31db9352a60ca5d3445e8ad9e72b0182099b

File: tv_apps/smart-home/js/home.js [hunk 1]
```diff
@@ -597,6 +650,57 @@
...
+                this.skipFolderBubble = undefined;
+              }.bind(this));
+
+            window.requestAnimationFrame(initFolderAnimation);
+          } else {
+            // 2nd frame, recover original transition.
+            this.folderListElem.style.transition = '';
+          }
+        }.bind(this);
+        var step = 0;
```
Comment (dwi2) [inline tv_apps/smart-home/js/home.js]: nit: I think it is more clear to put declaration of `step` before using it.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=66009d03a1466daeff3905de0983498b50db22d7

**Comment by rexboy7** [comment]
I just updated the patch at another commit such that edit mode also works.

Please take a look. Thanks!

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=227c3564d9109f9063ebc1cd6dfad7f2a32b7f4f

**Pull request merged**